### ISSUE TITLE
Number of elements does not match error for the test_data set

### ISSUE THREAD

**Issue by DR-LHX** [AI4EPS/PhaseNet#78]
When I tried the test_data sac.csv and sac folder files, I encountered the following error:

_tensorflow.python.framework.errors_impl.InvalidArgumentError:Cannot add tensor to the batch: number of elements does not match. Shapes are: [tensor]: [18001,1,3],batch]: [180001,3]_

How to solve this error problem? Is it because the input sac file contains only 18000 points?

Another question is whether this trained model fits for other seismic waveform containing different numbers of data points? I read from the paper that the model is trained under the waveform of 3000 points(30s and 100Hz)?

**Comment by napatt2** [comment]
Maybe you could add --batch_size=1.

**Comment by DR-LHX** [comment]
> Maybe you could add --batch_size=1.

Thanks very much for the reply. This suggestion works fine and right now I can run the test_data set. Another question is that the test data (waveform) has a length of 18000 points. The trained model is based on data sets with 3000 points and my data set has another different number of points in each seismic file. Can the prediction model still apply to my datasets and should I change the format of the datasets?

**Comment by zhuwq0** [comment]
Thanks @napatt2. Yes, you can apply to a long sequence of data @DR-LHX .

**Comment by DR-LHX** [comment]
> Thanks @napatt2. Yes, you can apply to a long sequence of data @DR-LHX .

Thanks, Dr. Zhu. May I ask if the model can be applied to 1 component seismic data, e.g. data contains z direction only. I saw the training data and test data in the model contain 3 components.

**Comment by zhuwq0** [comment]
Yes, you can use 1 component data, but the performance will reduce, particular for S wave.
